nemo/src/builder_proxy.rs

```diff
@@ -11,286 +18,491 @@ use nemo_physical::{
+    String(LogicalStringColumnBuilderProxy<'a, 'b>),
+    Integer(LogicalIntegerColumnBuilderProxy<'b>),
+    Float64(LogicalFloat64ColumnBuilderProxy<'b>),
+}
+///
 /// [`LogicalColumnBuilderProxy`] to add Any
 #[derive(Debug)]
 pub struct LogicalAnyColumnBuilderProxy<'a: 'b, 'b> {
-    physical: &'b mut PhysicalStringColumnBuilderProxy<'a>,
+    inner: &'b mut PhysicalStringColumnBuilderProxy<'a>,
 }
 
-impl LogicalAnyColumnBuilderProxy<'_, '_> {
-    fn normalize_string(input: String) -> String {
-        const BASE: &str = "a:";
-
-        let trimmed = input.trim();
-
-        if trimmed.is_empty() {
-            return r#""""#.to_string();
-        }
-
-        let data = format!("<> <> {trimmed}.");
-        let parser = TurtleParser::new(
-            BufReader::new(data.as_bytes()),
-            Iri::parse(BASE.to_string()).ok(),
-        );
-
-        let terms = parser
-            .into_iter(|triple| {
-                let normalized =
-                    TurtleEncodedRDFTerm::new(triple.object.to_string()).into_normalized_string();
-
-                Ok::<_, ReadingError>(match normalized.strip_prefix(BASE) {
-                    Some(stripped) => stripped.to_string(),
-                    None => normalized,
-                })
-            })
-            .collect::<Result<Vec<_>, _>>();
-
-        if let Ok(terms) = terms {
-            // make sure this really parsed as a single triple
-            if terms.len() == 1 {
-                return terms.first().expect("is not empty").to_string();
-            }
+impl<'a, 'b> LogicalAnyColumnBuilderProxy<'a, 'b> {
+    pub fn new(physical_builder_proxy: &'b mut PhysicalBuilderProxyEnum<'a>) -> Self {
+        match physical_builder_proxy {
+            PhysicalBuilderProxyEnum::String(inner) => Self { inner },
+            _ => unreachable!("If the database representation of the logical types is correct, we never reach this branch.")
         }
+    }
 
-        // not a valid RDF term.
-        // check if it's a valid bare name
-        if let Ok((remainder, _)) = parse_bare_name(span_from_str(trimmed)) {
-            if remainder.is_empty() {
-                // it is, pass as-is
-                return trimmed.to_string();
-            }
-        }
+    pub fn into_parser<Intermediate>(self) -> GenericLogicalParser<Intermediate, Self>
+    where
+        Self: ColumnBuilderProxy<Intermediate>,
+    {
+        GenericLogicalParser::new(self)
+    }
+}
 
-        // might still be a full IRI
-        if Iri::parse(trimmed).is_ok() {
-            // it is, pass as-is
-            return trimmed.to_string();
-        }
+impl ColumnBuilderProxy<Term> for LogicalAnyColumnBuilderProxy<'_, '_> {
+    logical_generic_trait_impl!();
 
-        // otherwise it needs to be quoted
-        format!(r#""{trimmed}""#).to_string()
+    fn add(&mut self, input: Term) -> Result<(), ReadingError> {
+        <Self as ColumnBuilderProxy<Term>>::commit(self);
```

Comment:
This doesn't need to commit.

Reply:
(Unless I missed something), I'm always committing manually if an error might occur before the inner `add` is called.
I think this is necessary to prevent rollbacks of input rows that have been read completely.
We could unify this to commit manually everywhere; wouldn't hurt.

Reply:
Ah, you're right, I had not though about errors.